dirty_cat/datasets/tests/test_fetching.py

```diff
@@ -1,238 +1,352 @@
-        os.remove(fil)
-    finally:
-        shutil.rmtree(test_dir)
 
+            # First, we want to purposefully test ValueError exceptions.
+            with pytest.raises(ValueError):
+                assert fetch_openml_dataset(dataset_id=0,
+                                            data_directory=test_data_dir)
+                assert fetch_openml_dataset(dataset_id=2**32,
+                                            data_directory=test_data_dir)
 
-@utils.with_setup(setup=setup_mock, teardown=teardown_mock)
-def test_convert_file_to_utf8(monkeypatch):
-    from dirty_cat.datasets import fetching
-    datadir = os.path.join(datasets_utils.get_data_dir(),
-                           'testdata')
-    try:
-        # Specify some content encoded in latin-1, and make sure the final file
-        # contains the same content, but in utf-8. Here, '\xe9' in latin-1 is
-        # '\xc3\xa9' in utf-8
-
-        with monkeypatch.context() as m:
-            m.setattr(utils.MockResponse, "zipresult", False)
-            m.setattr(utils.MockResponse, "_file_contents", b'\xe9')
-
-            dataset_info = fetching.DatasetInfo(
-                name='testdata',
-                urlinfos=(fetching.UrlInfo(
-                    url='http://foo/data',
-                    filenames=('data',),
-                    uncompress=False,
-                    encoding='latin-1'),),
-                main_file='data',
-                source='http://foo/')
-
-            info = fetching.fetch_dataset(dataset_info, show_progress=False)
-
-            with open(info['path'], 'rb') as f:
-                content = f.read()
-
-            assert content == b'\xc3\xa9'
-            os.unlink(info['path'])
-
-            m.setattr(utils.MockResponse, "zipresult", True)
-
-            dataset_info_with_zipfile = fetching.DatasetInfo(
-                name='testdata',
-                urlinfos=(fetching.UrlInfo(
-                    url='http://foo/data.zip',
-                    filenames=('unzipped_data.txt',),
-                    uncompress=True,
-                    encoding='latin-1'),),
-                main_file='unzipped_data.txt',
-                source='http://foo/')
-
-            info_unzipped = fetching.fetch_dataset(
-                dataset_info_with_zipfile, show_progress=False)
-
-            with open(info_unzipped['path'], 'rb') as f:
-                content_unzipped = f.read()
-
-            assert content_unzipped == b'\xc3\xa9'
+            # Valid call
+            returned_info = fetch_openml_dataset(dataset_id=test_dataset["id"],
+                                                 data_directory=test_data_dir)
+
+        except URLError:
+            # No internet connection, or the website is down.
+            # We abort the test.
+            #
+            # One could try to manually recreate the tree structure
+            # created by ``fetch_openml()``,  and the
+            # ``.gz`` files within in order to finish the test.
```

Comment:
I'm not sure that we should do this totally silently.

I think that we should:
1. Pop a warning (or something similar)
2. Skip using some of pytests' functionality for this (for instance calling pytest.skip)

Reply:
The goal is to have some track that the test is being skipped, and not silently loose track of it